www/js/controllers/route-controller.js

```diff
@@ -1,12 +1,24 @@
-angular.module('pvta.controllers').controller('RouteController', function($scope, $stateParams, Route, RouteVehicles, FavoriteRoutes){
+angular.module('pvta.controllers').controller('RouteController', function($scope, $stateParams, Route, RouteVehicles, FavoriteRoutes, Messages){
   var size = 0;
   var route = Route.get({routeId: $stateParams.routeId}, function() {
     route.$save();
     getHeart();
     $scope.stops = route.Stops;
     $scope.vehicles = route.Vehicles
+
+    // Need route to be defined before we can filter messages
+    var messages = Messages.query(function(){
+      var filteredMessages = [];
+      for(var message of messages){
+        if(message.Routes.indexOf($scope.route.RouteId) === -1) { continue; }
```

Comment:
Yep, that's true; I figured if we choose to implement any of the additional checks, then it'll be easy to incorporate it in the same way as a series of guards:

```js
if(message.Routes.indexOf($scope.route.RouteId) === -1) { continue; }
if(someOtherCheckFails) { continue; }
if(yetAnotherCheckFails) { continue; }
filteredMessages.push(message);
```